### ISSUE TITLE
Drop Unity Error captured by sentry.native.android, as default behavior

### ISSUE THREAD

**Issue by thinkocapo** [getsentry/sentry-java#3991]
## Steps to Repro
1. make an app with
https://github.com/Unity-Technologies/uaal-example/blob/master/docs/android.md. 
https://docs.unity3d.com/Manual/UnityasaLibrary.html. 
2. sentry demo app made via...
https://github.com/bitsandfoxes/uaal_wip
3. error when user leaves the Unity game
```
override funonDestroy(owner: LifecycleOwner) {
  unityPlayer.quit()
  super.onDestroy(owner)
}
```

### Solution Brainstorm

don't have this get captured as an error

**Comment by bruno-garcia** [comment]
@bitsandfoxes @romtsn  is this something you guys could add to that UaaL repro you built?

**Comment by bitsandfoxes** [comment]
>  is this something you guys could add to that UaaL repro you built?

Sure! I'll update here.

**Comment by bitsandfoxes** [comment]
Do I read that right, the issue is about dropping all captured native events during shutdown?

**Comment by bruno-garcia** [comment]
> Do I read that right, the issue is about dropping all captured native events during shutdown?

My understanding was that when the customer closed the Unity app, the native (Android) SDK captured an event that was not actionable. They indicated as if exiting Unity raised a signal that our SDK picked up. But they see that as "business as usual"

**Comment by bruno-garcia** [comment]
> > is this something you guys could add to that UaaL repro you built?
> 
> Sure! I'll update here.

Any luck with repro'ing this so we can decide next steps?

**Comment by markushi** [comment]
@bitsandfoxes is this something you want to take a look at?

**Comment by bruno-garcia** [comment]
> Do I read that right, the issue is about dropping all captured native events during shutdown?

@thinkocapo could you please confirm the behavior intended?

@bitsandfoxes can we reproduce this on the UaaL example we have? If so, what's the complexity of filtering out this event?

Internal customer case: https://www.notion.so/sentry/15e8b10e4b5d80a8bac6ff132144b8f8

**Comment by thinkocapo** [comment]
@bruno-garcia yes that's correct, they want to drop all captured native events that come from the shutdown code they provided. I updated the customer case with a screenshot of a particular one they ran into recently, as well as the issue link to it.

**Comment by bitsandfoxes** [comment]
We're going to pick this back up on the way of supporting UaaL properly as part of https://github.com/getsentry/sentry-unity/issues/1727
